xds/googledirectpath/googlec2p_test.go

```diff
@@ -180,9 +180,9 @@ func TestBuildXDS(t *testing.T) {
 
 			configCh := make(chan *bootstrap.Config, 1)
 			oldNewClient := newClientWithConfig
-			newClientWithConfig = func(config *bootstrap.Config) (xdsclient.XDSClient, error) {
+			newClientWithConfig = func(config *bootstrap.Config) (xdsclient.XDSClient, func(), error) {
 				configCh <- config
-				return tXDSClient, nil
+				return tXDSClient, func() { tXDSClient.Close() }, nil
```

Comment:
Do we not want to follow the same `close` pattern with the test xds client?  It seems that would be best, but maybe it's too much work or doesn't make sense for other reasons?

Reply:
I tried, but looks like a bit of work here. There is just way too much which is overridden in this test, and it would be nicer to not do that. But also, all this test wants to verify is that the client with built with the appropriate config and that is cleaned up at the end. It would definitely be nicer to have an e2e test though, but I guess the c2p resolver needs a lot of stuff (like running on GCE, DP being enabled etc).